Azure/azure-cli
# [DO NOT MERGE][ACR] az acr repository metadata show/update/delete: Add repository metadata API exposure.

**Description**  
Enables listing of metadata keys, downloading metadata by key to file, updating metadata by key from file, and deleting metadata by key. Metadata is associated with a repository, manifest, or tag.

**Testing Guide**  
Assuming your ACR registry is named `registryName` and it contains a repository named `repositoryName`:
1. List current metadata:
```
az acr repository metadata show --name registryName --repository repositoryName
```
2. Create a text file `testFile.in` in the current directory with some content. Upload it as metadata under the key `testKey`:
```
echo "test file contents" > testFile.in
az acr repository metadata update --name registryName --repository repositoryName --key testKey --file testFile.in
```
3. See that the repository now has metadata with key `testKey`:
```
az acr repository metadata show --name registryName --repository repositoryName
```
4. Download the metadata under key `testKey` to file `testFile.out` in the current directory; it should have the same content as `testFile.in`:
```
az acr repository metadata show --name registryName --repository repositoryName --key testKey --file testFile.out
cat testFile.out
```
5. Delete metadata under key `testKey`:
```
az acr repository metadata delete --name registryName --repository repositoryName --key testKey
```
6. See that the repository no longer has metadata with key `testKey`:
```
az acr repository metadata show --name registryName --repository repositoryName
```
7. The above steps manipulated metadata associated with the entire repository `repositoryName`. To manipulate metadata associated with the `tagName` tag, replace `--repository repositoryName` with `--image repositoryName:tagName`. To manipulate metadata associated with the manifest with digest `sha256:digest`, replace `--repository repositoryName` with `--image repositoryName@sha256:digest` instead.

**History Notes**

[ACR] az acr repository metadata show/update/delete: Add repository metadata API exposure.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
